### PR TITLE
Add `smt64` module to `stdlib` collections

### PR DESCRIPTION
**Pull request by bobbinth**
This PR implements `smt64` module in the standard library. This implementation of SMT is basically a map from a single-element key to a word value. For the rollup purposes, this could be used to represent account database as well as fungible asset vaults.

For now, the implementation is a thin wrapper around a `SimpleSmt` with depth 64. This is pretty inefficient as each lookup requires 64 hashes. We will improve this in the future by replacing `SimpleSmt` implementation with a compact Tiered Sparse Merkle tree with single-element keys.

- [x] Implement `get`, `set`, and `insert` procedures for `smt64`.
- [x] Write tests.

### CODE REVIEW

**Review by Al-Kindi-0 — APPROVED** [review]
Looks good to me, thank you!

**Review by frisitano — APPROVED** [review]
Looks good!

File: stdlib/asm/collections/smt64.masm [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+#!
+#! Fails if the tree with the specified root does not exits in the VM's advice provider.
+export.set
+    # prepare the stack for mtree_set operation
+    movup.4 movdn.8 swapw movup.8 push.64
+    # => [64, key, ROOT, VALUE, ...]
+
+    mtree_set
+    # => [OLD_VALUE, NEW_ROOT, ...]
+end
```
[the remark this answers is not in the document]
Comment (bobbinth) [reply stdlib/asm/collections/smt64.masm:75]: The main motivation is that `set` should also handle `[ZERO; 4]` values and treat them as deletions. Deletions are likely to introduce complexity - so, we provide a simple insert (which could be used to update the map when we know values inserted are not `[ZERO; 4]`) but also provide a more complex `set` for when it is needed.

In the context of a rollup, we'd be able to use `insert` for such things as updating account and nullifier databases, but we'd need to use `set` for things like updating account storage and vault.

File: stdlib/asm/collections/smt64.masm [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        dup.3 eq.0
+    end
+    and and and assertz
+
+    # prepare the stack for mtree_set operation
+    movup.4 movdn.8 swapw movup.8 push.64
+    # => [64, key, ROOT, VALUE, ...]
+
+    mtree_set
+    # => [OLD_VALUE, NEW_ROOT, ...]
```
[the remark this answers is not in the document]
Comment (bobbinth) [reply stdlib/asm/collections/smt64.masm:51]: I kept them separate as in the future these would have different implementations.

**Pull request merged**